src/types/node.rs

```diff
@@ -0,0 +1,238 @@
+use std::sync::Arc;
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use super::util::*;
+use super::*;
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+pub struct Node {
+    pub node_id: NodeId,
+    pub input_pads: Option<Vec<NodeId>>,
+    pub fallback_id: Option<NodeId>,
+
+    #[serde(flatten)]
+    pub params: NodeParams,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case", deny_unknown_fields)]
+pub enum NodeParams {
+    WebRenderer(WebRenderer),
+    Shader(Shader),
+    Image(Image),
+    Text(Text),
+    Transition(Transition),
+    #[serde(rename = "builtin:transform_to_resolution")]
+    TransformToResolution(TransformToResolution),
```

Comment:
I did not intend to change that, if you suggest a change please explain what you would like to switch to. (I still plan to change internally, but I'm not sure about the API yet)

Reply:
I mean using `FitToResolution`, `FillToResolution`, and `StretchToResolution` instead of a single `TransformToResolution`, as I suggested on Thursday.

Reply:
ok we can discuss it, but either way it won't be done in this PR